temporalcli/internal/cmd/gen-docs/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/temporalio/cli/temporalcli/commandsgen"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	// Get commands dir
+	_, file, _, _ := runtime.Caller(0)
+	commandsDir := filepath.Join(file, "../../../../docs/")
+
+	// Parse markdown
+	cmds, err := commandsgen.ParseCommands()
+	if err != nil {
+		return fmt.Errorf("failed parsing markdown: %w", err)
+	}
+
+	// Generate docs
+	b, err := commandsgen.GenerateDocsFiles(cmds)
+	if err != nil {
+		return err
+	}
+
+	// Write
+	for filename, content := range b {
+		filePath := filepath.Join(commandsDir, filename+".mdx")
```

Comment:
Is there anything `.mdx` specific about these files, or can they just be `.md` for use by everyone viewing this repo and not just our documentation engine?

Alternatively, we should consider having the docs repo generate this and put the files in their repo instead of having docs files outside of docs repo (I wouldn't expect any other SDK project to have the docs.temporal.io docs inside their repo).

Alternatively, like other SDKs, we could consider a place for a more "API docs" type home. In the same way we have typescript.temporal.io, python.temporal.io, dotnet.temporal.io, we could have cli.temporal.io. We can have CI push to this automatically like we do in those other repos.

Reply:
I don't think so, we can name it `.md` here, and in the future PR where I automate pushing out a PR to the docs repo, I'll add logic to rename it to the `.mdx` that the doc engine expects. 

Good idea!

Reply:
Actually, GH renders .mdx in the same way it renders .md, so anyone viewing this repo will see the .mdx files as if they were .md files. Is there any other reason we'd want to use .md here?

Reply:
I'm checking with team to see if there's any issues delivery md and not mdx for these very flat files

Reply:
There are no known issues but no compelling reason to pull one section out of alignment with the rest of the docs site. Recommend keep as mdx.

Reply:
> Is there any other reason we'd want to use .md here?

GH is not the only renderer/viewer of these files. I think unless it uses mdx-specific features, we might as well make it just md. Can rename for docs engine specifically of course.

Or maybe even better, don't commit the generated files in this repo if they are only used for the docs repo. Rather, have the generator here and have the docs engine/generator/CI/whatever run the generator and point to a dir they want to put the mdx files instead of assuming they should be in a `docs/` folder here.

Reply:
good idea, that will make contributions in this space easier, allowing the CI to handle everything and generate a docs PR without requiring the user to run an additional command to build the docs in this repo